lightning/src/ln/channelmanager.rs

```diff
@@ -4498,12 +4589,91 @@ impl_writeable_tlv_based!(HTLCPreviousHopData, {
 	(6, incoming_packet_shared_secret, required)
 });
 
-impl_writeable_tlv_based!(ClaimableHTLC, {
-	(0, prev_hop, required),
-	(2, value, required),
-	(4, payment_data, required),
-	(6, cltv_expiry, required),
-});
+impl Writeable for ClaimableHTLC {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		let payment_data = match &self.claim_type {
+			ClaimableType::Invoice(data) => data.clone(),
+			_ => msgs::FinalOnionHopData {
+				payment_secret: PaymentSecret([0; 32]),
```

Comment:
Why write a dummy `payment_data`? Best to just make the `payment_data` an `option` field, then old clients will immediately barf when they see it missing (which I believe is correct).

Reply:
I thought if we're making old clients barf we need a version bump and a whole MIN_VERSION thing? :thinking:

Reply:
Hmm, I guess it'll provide a better error message instead of just "InvalidValue", but I don't see why that's super critical. As long as its in the release notes.